yarl/_url.py

```diff
@@ -233,6 +233,8 @@ def build(
             raise ValueError(
                 'Can\'t mix "authority" with "user", "password", "host" or "port".'
             )
+        if not isinstance(port, (int, None.__class__)):
```

Comment:
A more idiomatic way would be
```suggestion
        if not isinstance(port, (int, type(None))):
```